src/librustc/hir/intravisit.rs

```diff
@@ -396,9 +400,8 @@ pub fn walk_mod<'v, V: Visitor<'v>>(visitor: &mut V, module: &'v Mod, mod_hir_id
 }
 
 pub fn walk_body<'v, V: Visitor<'v>>(visitor: &mut V, body: &'v Body) {
-    for argument in &body.arguments {
-        visitor.visit_id(argument.hir_id);
-        visitor.visit_pat(&argument.pat);
+    for arg in &body.arguments {
```

Comment:
```suggestion
    walk_list!(visitor, visit_arg, &body.arguments);
```
visitor.rs and intravisit.rs have this macro for visiting anything iterable like lists or options, it's used almost everywhere.